### PR TITLE
go.mod: bump most deps for start of cycle

### PR DESCRIPTION
**Pull request by raggi**
Plan9 CI is disabled. 3p dependencies do not build for the target.
Contributor enthusiasm appears to have ceased again, and no usage has
been made.

Skipped gvisor, nfpm, and k8s.

Updates https://github.com/tailscale/tailscale/issues/5794
Updates https://github.com/tailscale/tailscale/issues/8043

### CODE REVIEW

**Review by WalterHub — APPROVED** [review]

**Review by danderson — APPROVED** [review]

**Comment by danderson** [comment]
For the record, discussion elsewhere: the plan9 port is currently stalled on a crash in the Go runtime, which explains the lack of visible progress in our repo. Sounds like we're still happy to accept a maintainer's patches for plan9, but at present it's blocked on other things and doesn't work yet, so we're going to turn off that CI until the port is ready to resume.

**Pull request merged**
